lib/Backends/NNPI/NNPI.cpp

```diff
@@ -308,6 +330,23 @@ bool NNPIBackend::shouldLower(const Node *N) const {
   case Kinded::Kind::BatchedReduceMeanNodeKind:
   case Kinded::Kind::BatchedReduceMinNodeKind:
     return false;
+  case Kinded::Kind::FusedRowwiseQuantizedSparseLengthsSumNodeKind: {
+    const FusedRowwiseQuantizedSparseLengthsSumNode *SLSN =
+        llvm::cast<FusedRowwiseQuantizedSparseLengthsSumNode>(N);
+    if ((backendOptions_.useIceT || backendOptions_.inferOnDevice) &&
+        (SLSN->getResult().getElementType() == ElemKind::Float16Ty)) {
+      return false; // Don't lower == keep without weights
+    } else {
+      return true;
+    }
+  }
+  case Kinded::Kind::SparseLengthsSumNodeKind:
+    // WA - lower until ICE-T implements it.
```

Comment:
So this is temporary because SLS is implemented in ICEREF but still waiting on the implementation to be ready for device, right?

Reply:
@jfix71 You are correct.